### DISCUSSION TITLE
Quantification based exclusively on proteotypic peptides for mixed species samples?

### DISCUSSION THREAD

**Discussion by vincentroyrichard** [vdemichev/DiaNN#1006] [Q&A]
Is there is a way to have protein level quantitiation based exclusively on proteotypic peptides in DIA-NN? I am analyzing samples from mixed species (for example mouse and human) using a spectral library generated in MSFragger from both organisms and would like to make sure that shared peptides between organisms are not used for protein inference or quantitation. 

Thanks very much,
Vincent

**Comment by vdemichev** [comment]
Hi Vincent,

Yes, can use Genes.MaxLFQ.Unique for this.

- With the MSFragger library, please rename the column 'Proteotypicity' to 'Proteotypic' in the library, then should work.
One caveat specifically with human & mouse, since gene names are shared, makes sense to also edit the Genes column in the library to add a species suffix.

- Alternatively to the above, use the 'Reannotate' option in DIA-NN with the respective spectral libraries and set protein inference to 'Genes species-specific'.

Best,
Vadim

**Reply by vincentroyrichard** [reply]
Thanks for the information Vadim! I will give this a shot.

**Reply by vincentroyrichard** [reply]
Hi Vadim,

Is the "report.unique_genes_matrix.tsv" file giving the same information? Or is there a way to specify the export of the Genes.MaxLFQ.Unique column as a matrix similar to report.pg_matrix.tsv from the prompt in DIA-NN?

Thanks again for your help with this,
Vincent

**Reply by vdemichev** [reply]
Yes, unique_genes matrix contains the quantities corresponding to Genes.MaxLFQ.Unique.
You can obtain a 'matrix' like representation using the diann_matrix function of the diann R package. Or sth like this:
```
library(dcast)
library(reshape2)
pivot <- function(df, file.header, id.header, quantity.header) {
  df <- unique(df[,c(file.header, id.header, quantity.header)])
  melted <- as.data.frame(melt(df, id.vars = c(file.header, id.header), measure.vars = c(quantity.header)))
  melted$value[which(melted$value == 0)] <- NA
  piv <- dcast(melted, as.formula(paste0(id.header,'~',file.header)), value.var = "value") 
  rownames(piv) <- piv[[1]]
  piv[,1] <- NULL
  as.matrix(piv)
}
```

Best,
Vadim
